lib/xandra/protocol.ex

```diff
@@ -26,6 +26,20 @@ defmodule Xandra.Protocol do
     end
   end
 
+  defmacrop decode_metadata({:<-, _, [value, buffer]}) do
+    quote do
+      {unquote(value), unquote(buffer)} =
+        case Application.get_env(:xandra, :metadata_format, :string) do
```

Comment:
Regardless if we're gonna merge this, but in the meantime: doing this means doing an ETS lookup *every time* we need to decode a map key. We need to pass this down from outside as an argument if we're going to do this. Also, I wouldn't configure this at the application level (for `:xandra`), but rather at the connection level (in `start_link`, as an option).

Reply:
I was going to ask about the implementation, I'm not wedded to it. I'm assuming the app variables are set in ETS, hence your comment.

I went back and forth between a connection-level vs. app level setting; I came down on app-level because I could see one half of your app accidentally using atoms and the other half using strings. I don't feel too strongly about it, and am happy to change it around.